arionBackend/transformation/esper/esper_query.py

```diff
@@ -0,0 +1,155 @@
+		:param event_type: the event type to inspect
+		:return: updated event type
+		"""
+		try:
+			tokenized_event_type = parse("{name}.win:{type}", event_type["event"]["name"]).named
+			event_type["event"]["name"] = tokenized_event_type["name"]
+			if tokenized_event_type["type"] != "keepall()":
+				tokenized_type = parse("{type}({size})", tokenized_event_type["type"]).named
+				tokenized_with_window = {
+					"window": {"event": event_type["event"]},
+					"type": tokenized_type["type"],
+					"size": tokenized_type["size"]
+				}
+			else:
+				tokenized_with_window = {"window": {"event": event_type["event"]}, "type": tokenized_event_type["type"]}
+			if "alias" in event_type:
+				tokenized_with_window["alias"] = event_type["alias"]
+			return tokenized_with_window
+		except AttributeError:
+			return event_type
+
+	@staticmethod
+	def tokenize_input_stream(input_stream):
+		"""
+		Tokenizes a given input stream, looks for aliases, selections and windows.
+		:param input_stream: the input stream to inspect
+		:return: the tokenized input stream
+		"""
+		tokenized_input = QueryParser.find_alias(input_stream)
+		tokenized_input["event"] = QueryParser.find_selection(tokenized_input["event"])
+		return QueryParser.find_window(tokenized_input)
+
+	@staticmethod
+	def parse_input_clause(input_clause):
+		"""
+		Parses the input clause: Single Event Types, Joins and Patterns are possible.
+		:param input_clause: the input clause to parse
+		:return: the updated input clause
+		"""
+		updated_input_clause = {"input": {}}
+		if "," in input_clause:
+			# if a comma is in the input clause, multiple event streams are joined as input
+			stripped_event_streams = [event_stream.strip() for event_stream in input_clause.split(',')]
+			updated_input_clause["input"]["data"] = \
+				[QueryParser.tokenize_input_stream(event_stream) for event_stream in stripped_event_streams]
+			input_type = "JOIN"
+		else:
+			try:
+				tokenized_pattern = parse("pattern {pattern}", input_clause).named
+				updated_input_clause["input"]["data"] = \
+					QueryParser.parse_pattern(tokenized_pattern["pattern"].strip())
+				input_type = "PATTERN"
+			# if no error is thrown, the pattern clause was detected
+			except AttributeError:
+				# input must be single event type
+				updated_input_clause["input"]["data"] = input_clause.strip()
+				updated_input_clause["input"]["data"] = \
+					QueryParser.tokenize_input_stream(updated_input_clause["input"]["data"])
+				input_type = "SINGLE"
+		updated_input_clause["input"]["type"] = input_type
+		return updated_input_clause
+
+	@staticmethod
+	def parse_pattern(pattern):
+		"""
+		Parses the pattern from the input clause.
+		:param pattern: the pattern to parse
+		:return: eqmn representation of the pattern
+		"""
+		pattern = pattern.strip("[").strip("]")
+		# TODO: Implement pattern parsing (this is not necessary for the prototypical implementation, IMHO)

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Add a citation of the person who can best explain this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.tommartensen.de/coding_rules#rule_key=python%3AS1707)